scene/animation/animation_player.cpp

```diff
@@ -664,7 +664,32 @@ void AnimationPlayer::_animation_process_animation(AnimationData *p_anim, double
 					}
 				}
 
-				if (update_mode == Animation::UPDATE_CONTINUOUS || update_mode == Animation::UPDATE_CAPTURE || (p_delta == 0 && update_mode == Animation::UPDATE_DISCRETE)) { //delta == 0 means seek
+				if (update_mode == Animation::UPDATE_DISCRETE) {
```

Comment:
You must use `else if` because track will never have more than one `update_mode` at the same time. Multiple `if`s cause unnecessary evaluations.

Reply:
There is an `if` block above for the CAPTURED mode, and below there is an `if` (not `else if`) block, which (together with it's `else if` clause) catches all the modes. So, I I assumed that this `if` block above is a special case for CAPTURED mode, and I can add another one for the DISCRETE mode. Then, the block below will be executed for all the modes, to handle the rest of the processing (when special cases have already been handled).

Could you please verify if this works like this? If not, then I assume I should also change the block below to `else if`, so it would be:

```
if (update_mode == Animation::UPDATE_CAPTURE) {
...
} else if (update_mode == Animation::UPDATE_DISCRETE) {
...
} else if (update_mode == Animation::UPDATE_CONTINUOUS || update_mode == Animation::UPDATE_CAPTURE) {
...
} else if (p_is_current && p_delta != 0) {
...
}
```

Reply:
@TokageItLab  It took me a moment to realize that you probably mean just the expression I've added :) I've changed it to `else if` to the previous handling of CAPTURE mode, and left the if/else if sequence below untouched. Is that OK?

My question, however, was about a different thing. I've noticed that there is usually an `else` clause for the `if (pa->accum_pass != accum_pass)` expression in other places here, but I couldn't figure out what I should put there for the DISCRETE mode. All the examples used value interpolation in the `else if`, but that's not the case for DISCRETE (in this mode, we don't interpolate). So, I've moved the assignment of the value outside of `if` whatsoever, and therefore there is no `else`. Is that OK?